### PR TITLE
Added type definitions for @vimeo/player

### PR DESCRIPTION
**Pull request by denisyilmaz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

references #15571

### CODE REVIEW

**Comment by andy-ms** [comment]
This should go in `types/vimeo__player`. See Microsoft/TypeScript#15051.

**Comment by denisyilmaz** [comment]
I tried as suggested. updated VSC Workspace to typescript 2.3.0. but scoped module is not found in my test.ts file. (with updated tsconfig to "baseUrl": "types" etc…).

**Comment by andy-ms** [comment]
That feature won't be out until 2.3.1. This could either wait, or use `declare module "@vimeo/player { }`.
This branch now contains commits that aren't yours. Please rebase to include just your changes. (Be sure to make a backup first.)

**Comment by denisyilmaz** [comment]
thanks for the help. hope now it is as needed.

File: vimeo__player/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// Type definitions for @vimeo/player v2.0.1
+// Project: https://github.com/vimeo/player.js
+// Definitions by: Denis Yılmaz <https://github.com/denisyilmaz>, Felix Albert <f.albert.work@icloud.com>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3.1
```
Comment (andy-ms) [inline vimeo__player/index.d.ts]: This should just be `2.3`, and we'll always use the greatest minor version.

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   },
   "devDependencies": {
     "dtslint": "Microsoft/dtslint#production",
-    "types-publisher": "Microsoft/types-publisher#production"
+    "types-publisher": "Microsoft/types-publisher#production",
+    "typescript": "^2.3.0-dev.20170418"
```
Comment (andy-ms) [inline package.json]: Don't modify this file.

File: vimeo__player/@vimeo-player-tests.ts [hunk 3]
```diff
@@ -0,0 +1,456 @@
+import { Player } from '@vimeo/player'
```
Comment (andy-ms) [inline vimeo__player/@vimeo-player-tests.ts]: This should be in `types/vimeo__player/vimeo__player-tests.ts`.

Comment (denisyilmaz) [inline vimeo__player/@vimeo-player-tests.ts]: sorry for confusion, just to be sure: 
only the `vimeo__player-tests.ts` should be in the `types/vimeo__player` folder? 
the `vimeo__player/index.d.ts` file stays in the main folder? It was suggested in a "contributor handbook", so I'm unsure whats the right way.

Comment (andy-ms) [inline vimeo__player/@vimeo-player-tests.ts]: Everything should go in `types/vimeo__player`.

**Comment by denisyilmaz** [comment]
travis failed: `Could not parse version: line is '// TypeScript Version: 2.3'`
should i just delete it for now till typescript 2.3 is out?

**Comment by andy-ms** [comment]
Yes, and be sure to run `npm run lint vimeo__player`.

**Comment by denisyilmaz** [comment]
hm, with `npm run lint vimeo__player` getting following error:
```
Warning: Text includes DefinitelyTyped link, but '--dt' is not set.
Failed to load /Users/[…]/types/vimeo__player/tslint.json: Invalid "extends" configuration value - could not require "dtslint/dt.json". Review the Node lookup algorithm (https://nodejs.org/api/modules.html#modules_all_together) for the approximate method TSLint uses to find the referenced configuration file.
```

**Comment by andy-ms** [comment]
Looks like you're using an older version of dtslint, just reinstall.

**Comment by denisyilmaz** [comment]
ok, reinstalled, but getting following error `Object.entries is not a function`

**Comment by andy-ms** [comment]
Upgrade your `node` version too. It should have warned about that on install.

**Comment by denisyilmaz** [comment]
I'm sorry for my inability to fix this. Finally resolved all errors except for:
`ERROR: 6:1  no-single-declare-module  File has only 1 module declaration — write it as an external module.`

**Comment by andy-ms** [comment]
Just add a `// tslint:disable-next-line no-single-declare-module` above the `declare module`, since you need it until 2.3.1 is published.

File: types/vimeo__player/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
+// Type definitions for @vimeo/player 2.0
+// Project: https: //github.com/vimeo/player.js
```
Comment (andy-ms) [inline types/vimeo__player/index.d.ts]: Extra space

File: types/vimeo__player/index.d.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // Type definitions for @vimeo/player 2.0
-// Project: https: //github.com/vimeo/player.js
-// Definitions by: Denis Yılmaz <https: //github.com/denisyilmaz>, Felix Albert <f.albert.work@icloud.com>
-// Definitions: https: //github.com/DefinitelyTyped/DefinitelyTyped
+// Project: https://github.com/vimeo/player.js
+// Definitions by: Denis Yılmaz <mail@denisyilmaz.de>, Felix Albert <f.albert.work@icloud.com>
```
Comment (andy-ms) [inline types/vimeo__player/index.d.ts]: Actually, a github username is better so we can notify you an issues and pull requests related to this package.

**Comment by andy-ms** [comment]
Thanks!

**Comment by denisyilmaz** [comment]
One last question: I tried adding the typings to my recent angular project. the types get recognized just fine, but on compiler time i get the `/node_modules/@vimeo/player/dist/player.js' implicitly has an 'any' type.` error… 
should i open a new issue?

**Comment by andy-ms** [comment]
It looks like module resolution for an import in your code resolved to a JS file and didn't find any declaration to resolve to. 
If you tried importing e.g. `import * as p from "@vimeo/player/dist/player`", that would explain the problem since the type declaration here doesn't define such a module.

**Comment by denisyilmaz** [comment]
unfortunalty no. I'm importing the same way as in my test.ts file. `import { Player } from '@vimeo/player';`
VSC finds the installed typings correctly. I even added the "vimeo__player" to the compilerOptiony >> types.
`Could not find a declaration file for module '@vimeo/player'. '/Users/[…]/frontend/node_modules/@vimeo/player/dist/player.js' implicitly has an 'any' type.`

**Comment by denisyilmaz** [comment]
ok, i looked at another type-definition and they made it like this:

```typescriptdeclare namespace Player { }```

```typescript
declare module '@vimeo/player' {
	export = Player;
}

declare module '@vimeo/player/dist/player' {
	export = Player;
}
```

**Comment by andy-ms** [comment]
What other type definition? :confused: 
Try turning on `--traceResolution` to see where `/node_modules/@vimeo/player/dist/player.js` is getting imported in your code.

**Comment by denisyilmaz** [comment]
in the @types/mapbox-gl they declare it this way

**Comment by andy-ms** [comment]
Oh, thought you meant another definition for the same module.
For `mapbox-gl` they must have intended for someone to be able to `import * as gl from "mapbox-gl/dist/mapbox-gl"`, but if you're not actually directly importing from there it shouldn't be relevant...

**Comment by denisyilmaz** [comment]
Oh, no sorry for confusion. :)

i tried the traceResolution option, but it just shows the error in the component where I'm importing the "@vimeo/player"
```
Error at /Users/…/frontend/src/app/shared/vimeo-video/vimeo-video.component.ts:4:24: Could not find a declaration file for module '@vimeo/player'. '/Users/…/frontend/node_modules/@vimeo/player/dist/player.js' implicitly has an 'any' type.
Error at /Users/…/frontend/aot/src/app/shared/vimeo-video/vimeo-video.component.ngfactory.ts:12:26: Could not find a declaration file for module '@vimeo/player/dist/player'. '/Users/…/frontend/node_modules/@vimeo/player/dist/player.js' implicitly has an 'any' type.
Error at /Users/…/frontend/aot/src/app/page/main/element/element.component.ngfactory.ts:16:26: Could not find a declaration file for module '@vimeo/player/dist/player'. '/Users/…/frontend/node_modules/@vimeo/player/dist/player.js' implicitly has an 'any' type.
```

I'm nowhere declaring `/dist/player.js` to be imported, so it must happen in-between.
the only place I find `/dist/player.js` is in the plugin itself in the package.json '"types": "dist/player"'


```json
{
  "compilerOptions": {
    "target": "es5",
    "module": "es2015",
    "moduleResolution": "node",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": ["es2015", "dom"],
    "noImplicitAny": true,
    "suppressImplicitAnyIndexErrors": true,
    "types": [
      "node",
      "clipboard",
      "vimeo__player"
    ]
  },

  "files": [
    "src/app/app.module.ts",
    "src/main-aot.ts"
  ],

  "angularCompilerOptions": {
   "genDir": "aot",
   "skipMetadataEmit" : true
 }
}
```

**Comment by andy-ms** [comment]
It looks like module resolution is correctly resolving it as a JS module, but only because it didn't find the types in the first place. Are you sure you've correctly included those types? `--listFiles` would show you if it's lincluded.

**Comment by rugdealer** [comment]
I'm running an Ionic Project with Angular2/Typescript2 and I'm unable to import:
import {Player} from "@vimeo/player"

**Package.json**

Somewhere in my package.json
"dependencies": {
        "@types/vimeo__player": "^2.0.0",
        "@vimeo/player": "^2.1.0"
}

I have both installed.

**In one of my components:**

```
import {Component} from '@angular/core';
import {NavController} from "ionic-angular/index";
import {Player} from "@vimeo/player";


@Component({
  selector: 'player-component',
  templateUrl: 'player.html'
})
export class PlayerComponent {

  private player: Player;

  constructor(public navCtrl: NavController){

    this.player = new Player('handstick', {
      id: 173449968,
      width: 640
    });
  }
}
```

**Error:**

> Uncaught (in promise): TypeError: undefined is not a constructor (evaluating 'new __WEBPACK_IMPORTED_MODULE_2_vimeo_player__["Player"]('handstick', {id: 173449968, width: 640})')

**IDE: Webstorm**

Clearly shows `Player` and `player` in red from the `import` statement.

**Comment by andy-ms** [comment]
@rugdealer That looks like a run-time error, not a compile error.

**Comment by rugdealer** [comment]
@andy-ms it's true. It compiles correctly and crashes during run-time. I think if webstorm already notices an issue before hand (i.e.: Player and player in red) that there is an issue with the import (knowing that there are no issues with any other imports). Do you have any ideas what could be causing it? Thanks.

**Comment by denisyilmaz** [comment]
Actually we had similar problems with it.  I think the problem could be found here [https://github.com/vimeo/player.js#using-with-a-module-bundler](url) >> window.Vimeo

>> Is there a way we could add for  the Vimeo-Namespace?

The Vimeo-API Sets a global Vimeo Namespace in which the Player Class exist. Typings for Vimeo.Player would cause typescript compiler to fail with error, Typings for Player fail on runtime…

**Pull request merged**